dsgrid/dataset/pivoted_table.py

```diff
@@ -161,25 +164,32 @@ def process_pivoted_aggregations(
             if dimension_query_name is None:
                 continue
             dim_config = self.project_config.get_dimension(dimension_query_name)
+            dim_type = dim_config.model.dimension_type
+            if dimension_query_name not in supp_names:
+                raise Exception(f"Bug: {dimension_query_name=} is not a supplemental dimension")
             mapping_records = self.project_config.get_base_to_supplemental_mapping_records(
-                dimension_query_name
+                dim_config.model.dimension_query_name
             )
+            to_unit_records = dim_config.get_records_dataframe()
             df, new_pivoted_columns, dropped_columns = map_and_reduce_pivoted_dimension(
                 df,
                 mapping_records,
                 pivoted_columns,
                 agg.aggregation_function.__name__,
                 rename=False,
             )
-            dim_type = dim_config.model.dimension_type
-            if dim_type == DimensionType.METRIC:
-                df = energy.convert_units(
-                    df,
-                    new_pivoted_columns,
-                    self._project_config.get_base_dimension(dim_type).get_records_dataframe(),
-                    mapping_records,
-                    dim_config.get_records_dataframe(),
+            if dim_type != DimensionType.METRIC:
+                raise NotImplementedError(
+                    "Aggregation of a pivoted column that is not the metric dimension is not "
+                    f"yet supported: {dim_type}"
```

Comment:
Is this only because we are hard-coding somewhere to expect only metric-named columns?

Reply:
Unit conversion won’t work, at least.